widget/diagramwidget/geometry/r2/box.go

```diff
@@ -0,0 +1,175 @@
+package r2
+
+// Box defines a box in R2
+//
+//                A
+//                |
+//                |
+//                |
+//                v
+//    (1) A.X,A.Y +------+ A.X+S.X,A.Y (2)
+//                |\     |
+//                | \    |
+//                |  \   |
+//                |   \S |
+//                |    \ |
+//                |     \|
+// (3)A.X,A.Y+S.Y +------+ A.X+S.X,A.Y+S.Y (4)
+//
+type Box struct {
```

Comment:
Would it not be worth defining this as a `Box{fyne.Position, fyne.Size}`?
We have the types and maths in the core package

Reply:
The r2 package here is mostly about geometry operations. The algorithms in the package (e.g. what is the intersection between the box and a line) use these geometries. Not sure I want to re-write those algorithms at this time. On the other hand, I think it is appropriate to keep all of the r2 stuff local to the widget so that it can be refined/replaced later. I'll work on that.

Reply:
I don't see a practical way to limit the visibility of the r2 package components. I think we should leave it for now